libs/pageserver_api/src/key.rs

```diff
@@ -227,6 +227,9 @@ impl Key {
 // AuxFiles:
 // 03 00000000 00000000 00000000 00   00000002
 //
+// ReplOrigin:
+// 03 00000000 00000000 00000000 00   00000003
+//
```

Comment:
Let's not take this AuxFiles approach with a HashMap again. For all the same write amplification issues we had with auxfiles. Let's treat each replication origin as a separate kv-pair, and have one directory entry that lists them.

Reply:
I have thought a lot about it, but I think that in this case this approach is preferable.

We in any case have to store directory for replorigins, right?
In case of replication origins content (origin_id+origin_lsn) have almost the same size as just reference (origin_id).
So I do not see much sense in storing origin_lsn separately. It will not reduce speed but significantly increase time (because we have to fetch all elements of directory). 

Also number of replorigins is limited by `max_slot_number`  - it is 120 by default and in most case there will be no more than just one slot and replorigin.

Reply:
cc @skyzh. Note that this is different from storing the replication slots, which previously caused the massive write amplification issues. These replication origins are much simpler, but I can see similar danger here too.

Reply:
> The max number of slots you can have, if you bump up max_replication_slots, is 2^16.
Actually larger - `MAX_BACKENDS = 0x3FFFF`.
But please notice that still real number of origins is limited by number of replicas.
Even if you bump up `max_replication_slots`, still you will not get more different origins than LR channels.
And number of this channels is limited by max_connection, which is 100, but certainly can be increased for some customers.

In any case, I am 100% for doing it properly, especially after this amplification problem with replslots.
But I am still not quite sure what properly means in this case.
What is the status of AUX2 files support now?
As far as I see, it is not yet merged. If we do not have method for range search of our KV storage, then I do not see some better alternative to the current approach (with  ReplOriginDir).

If we still have some kind of "directory" object containing IDs of all entries (as we have for databases, relations, ...), 
then IMHO there is no sense in storing separate entries for replorigins, because size of content (origin_lsn = 8bytes) is comparable with ID (origin_id - 4 bytes).

And if we really want to use range scan for replorigins, then it seems to me that:
1. It is better to wait until AUX2 stuff will be committed (if it really implements range scan).
2. Think about refactoring of relation directory, because it seems to suffer from the same amplification problem much more than replorigins.

And last one my comment/concern about "proper implementation": if we will use range scan, then it means that we have to reconstruct N objects instead of 1. It obviously increase time of preparing of basebackup and so startup time. So there is obvious compromise between space (write amplification) and extraction speed. I think that we should take it in account choosing right approach.

Reply:
There is range scan. No directory keys. The last pull request in the queue will be merged today and I’ll have some perf improvement pull requests.

Reply:
I have changed this PR to store repl origins individually. and use range scan to fetch them.
@skyzh I will be pleased it you can review my changes (at least related with storing/retriving replorigins in KV storage). May be I missed something in how range scan works...